### DISCUSSION TITLE
feat: 增加单元化多活流量调度的支持

### DISCUSSION THREAD

**Discussion by hexiaofeng** [apache/apisix#7292] [Ideas]
### Description

单元化多活流量调度的需求，根据规则获取某个变量。
1）判断变量是否在单元的白名单里面，是则路由到指定单元
2）判断变量是否在单元的前缀白名单里面，是则路由到指定单元
3）采用BKDRHash算法获取变量hash值，并按照指定参数取模，判断结果区间在哪个单元，并路由到指定单元。
路由纠错，如果发现流量不是本单元的，则可以代理或重定向到正确的单元
单元的域名可能加了前缀，例如主域名是x.y.z，北京单元的域名是bj-x.y.z

能否完善traffic_split，支持单元化多活流量调度

**Comment by tzssangglass** [comment]
It seems that this is private requirements.

From your description, 1 & 2 can be done with `vars` in `traffic_split ` plugin.

3. It looks like BKDRHash needs to be implemented on Lua.

> 路由纠错，如果发现流量不是本单元的，则可以代理或重定向到正确的单元

How to find out that the traffic is not from this unit?

And in your description, the unit may be a term in your business context, could it be equivalent to APISIX upstream?

**Reply by hexiaofeng** [reply]
目前多个互联网大厂提供了单元多活的能力，并且行业也在制定相关的标准，思路是一致，参考大家的经验提的需求。

应用部署在Region-bj（AZ1，AZ2），Region-gz（AZ3，AZ4），按用户进行划分接入不同的区域，数据通过DTS进行同步。

有多层路由，边缘网关/全局网关根据1,2,3进行分发，后端单元网关会根据1,2,3判断是否是当前单元，如果不是会纠错。

unit不等于upstream，路由的时候，可能是通过代理或重定向到另外的单元域名。例如边缘网关 x.y.z -> bj-x.y.z（代理）
